ChiModules/LinearBoltzmannSolver/IterativeOperations/lbs_setsource.cc

```diff
@@ -171,15 +171,15 @@ SetSource(LBSGroupset& groupset,
             //============================== Loop over groups
             for (size_t gprime = first_grp; gprime <= last_grp; ++gprime)
             {
-              if ((gprime >= gs_i) or (gprime <= gs_f))
+              if ((gprime >= gs_i) and (gprime <= gs_f))
```

Comment:
@Naktakala @dominiccaron @wdarylhawkins @zachhardy Correct me if what I am saying here is wrong. If we do not have enough coverage through tests to catch this, we should reconsider the pace how which we are doing PRs. Shouldn't this have been caught by a test? We need more tests with our PRs, no?

Reply:
Valid point: the implementation of new functionalities should be accompanied by corresponding regression tests (of course, there may be existing functionalities in the code base that require regression tests as well). In this particular instance, the existing mono-energetic test of the eigenvalue solver is unable to detect a multi-group issue or a group-set issue, so a new test of the eigenvalue solver with multi-group and group-sets truly is merited. Likely, Zach has an input readily available for such a test.

Reply:
This is a verification test problem on a new feature not a regression test problem. It basically means that little to no verification test has been applied to multigroup problems. We can slow down PRs all we like but we can only walk by the light that shines at our feet. Additionally, if you slow down PRs then they tend to grow and diverge from the code base.

There was some value in saying that the keigensolver was a core module of ChiTech (mostly to maintain a sort of open-source status). I am making the motion now to freeze it as a little code demo and separate this module until the time that we can supply verification suites.